pkg/handlers/dashboard.go

```diff
@@ -24,15 +30,82 @@ func (h *Handler) GetAppDashboard(w http.ResponseWriter, r *http.Request) {
 	clusterID := mux.Vars(r)["clusterId"]
 	appStatus := new(appstatetypes.AppStatus)
 	if util.IsHelmManaged() {
-		appStatus.State = "ready"
-		getAppDashboardResponse := GetAppDashboardResponse{
-			AppStatus:         appStatus,
-			Metrics:           nil,
-			PrometheusAddress: "",
+		release := helm.GetHelmApp(appSlug)
+		if release == nil {
+			logger.Errorf("release for app: %s does not exist\n", appSlug)
+			w.WriteHeader(http.StatusNotFound)
+			return
 		}
 
-		JSON(w, 200, getAppDashboardResponse)
-		return
+		// we have received informer data
+		if len(release.Status.ResourceStates) > 0 {
+			getAppDashboardResponse := GetAppDashboardResponse{
+				AppStatus:         &release.Status,
+				Metrics:           nil,
+				PrometheusAddress: "",
+			}
+
+			JSON(w, 200, getAppDashboardResponse)
+			return
+		} else {
+			// we must apply the informers and await result
+			app := &apptypes.App{
```

Comment:
This whole block that waits for informer results should not be in an API handler.  If we are starting up the operator in the `Start()` function above, it should be doing this work asynchronously.